core/model/src/test/java/org/eclipse/sw360/antenna/model/test/ArtifactUtilPurlTest.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) Bosch Software Innovations GmbH 2019.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.sw360.antenna.model.test;
+
+import org.eclipse.sw360.antenna.model.artifact.facts.ArtifactCoordinates;
+import org.eclipse.sw360.antenna.model.artifact.facts.dotnet.DotNetCoordinates;
+import org.eclipse.sw360.antenna.model.artifact.facts.java.BundleCoordinates;
+import org.eclipse.sw360.antenna.model.artifact.facts.java.MavenCoordinates;
+import org.eclipse.sw360.antenna.model.artifact.facts.javaScript.JavaScriptCoordinates;
+import org.eclipse.sw360.antenna.model.util.ArtifactUtils;
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ArtifactUtilPurlTest {
+    @Test
+    public void testPurlParserWithMavenCoordinates() {
+        String purl = "pkg:maven/com.something/foo@1.2.3";
+
+        ArtifactCoordinates coordinates = ArtifactUtils.createArtifactCoordinatesFromPurl(purl);
+
+        assertThat(coordinates).isInstanceOf(MavenCoordinates.class);
+        assertThat(coordinates.getPurl().canonicalize()).isEqualTo(purl);
+    }
+
+    @Test
+    public void testPurlParserWithP2Coordinates() {
+        String purl = "pkg:p2/com.something.foo@1.2.3";
+
+        ArtifactCoordinates coordinates = ArtifactUtils.createArtifactCoordinatesFromPurl(purl);
+
+        assertThat(coordinates).isInstanceOf(BundleCoordinates.class);
+        assertThat(coordinates.getPurl().canonicalize()).isEqualTo(purl);
+    }
+
+    @Test
+    public void testPurlParserWithNugetCoordinates() {
+        String purl = "pkg:nuget/foo-dll@1.2.3";
+
+        ArtifactCoordinates coordinates = ArtifactUtils.createArtifactCoordinatesFromPurl(purl);
+
+        assertThat(coordinates).isInstanceOf(DotNetCoordinates.class);
+        assertThat(coordinates.getPurl().canonicalize()).isEqualTo(purl);
+    }
+
+    @Test
+    public void testPurlParserWithJavaScriptCoordinates() {
+        String purl = "pkg:npm/%40angular/animation@1.2.3";
```

Comment:
Not a part of this PR but a problem I have with purl: i dislike how purl handles the escaping of `@` as `%40` here. I would expect an URL to be stable under URL de- and en-coding (since some utils might use it).

Reply:
Oh sorry, that is absolutely correct and matches the spec of URL-encoding